### PR TITLE
Net API v2

### PR DESCRIPTION
**Pull request by pwalski**
resolves: https://github.com/golemfactory/yagna/issues/1882

### CODE REVIEW

**Comment by pwalski** [comment]
resolves: https://github.com/golemfactory/yagna/issues/1882

**Comment by pwalski** [comment]
I added v2 of /net (vpn) API specs alongside with v1. No idea right now if it is ok.

File: specs/net-api-v2.yaml [hunk 1]
```diff
@@ -0,0 +1,414 @@
...
+  - name: requestor
+    description: Requestor part
+
+paths:
+  /net/info:
+    get:
+      tags:
+        - provider
+      operationId: getInfo
+      summary: GetInfo - Fetches Provider connection info.
```
Comment (nieznanysprawiciel) [inline specs/net-api-v2.yaml:23]: For both Provider and Requestor

**Comment by pwalski** [comment]
After implementing it I am keen to change API from
`/net-api/v2/net/info`
`/net-api/v2/vpn/net`(etc.)
into:
`/net-api/v2/info`(or better `/net-api/v2/status`)
`/net-api/v2/vpn/net`
Having two top level resources of `net` and `vpn` is weird. In my mind this root `net` means nothing.

File: src/net.rs [hunk 2]
```diff
@@ -7,24 +7,52 @@ use awc::ws::Codec;
...
+        NetApi { client }
+    }
+}
+
+impl NetApi {
+    /// Retrieves connection info.
+    pub async fn get_info(&self) -> Result<Info> {
+        self.client.get("info").send().json().await
+    }
+}
```
Comment (nieznanysprawiciel) [inline src/net.rs:34]: `info` is very general name. I would consider `status`.
Check result of command `yagna net status`. I think we could return the same information.
(You can look at `StatusResponse` in `yagna/core/model/src/net`)

File: specs/net-api.yaml [hunk 3]
```diff
@@ -2,9 +2,7 @@ openapi: 3.0.2
 info:
   version: 0.1.0
   title: Yagna Net API
-  description: '
```
Comment (nieznanysprawiciel) [inline specs/net-api.yaml:5]: Is there any way to deprecate this old API?

File: src/net.rs (hunk 2 above)
Comment (pwalski) [reply src/net.rs:34]: Agree. Wrote just before in comment https://github.com/golemfactory/ya-client/pull/121#issuecomment-1168533694

File: specs/net-api-v2.yaml [hunk 4]
```diff
@@ -0,0 +1,414 @@
+openapi: 3.0.2
+info:
+  version: 0.1.0
```
Comment (pwalski) [inline specs/net-api-v2.yaml:3]: Should have probably bump it to 0.2.0

File: src/net.rs (hunk 2 above)
Comment (nieznanysprawiciel) [reply src/net.rs]: I'm against skipping it

**Review by nieznanysprawiciel — APPROVED** [review]

**Comment by nieznanysprawiciel** [comment]
After discussing topic with @stranger80, we decided that public IP information should be added to (not created yet) capabilities API: https://github.com/golemfactory/golem-architecture/pull/46
Rest of changes in this PR is ok.

Moved to issue:
https://github.com/golemfactory/yagna/issues/2066

**Pull request merged**